src/java/org/apache/cassandra/net/Message.java

```diff
@@ -717,7 +725,7 @@ private int serializedHeaderSizePost40(Header header, int version)
             long size = 0;
             size += sizeofUnsignedVInt(header.id);
             size += CREATION_TIME_SIZE;
-            size += sizeofUnsignedVInt(1 + NANOSECONDS.toMillis(header.expiresAtNanos - header.createdAtNanos));
+            size += sizeofUnsignedVInt(NANOSECONDS.toMillis(header.expiresAtNanos - header.createdAtNanos));
```

Comment:
Why aren't we adding the 1 to the serializeHeader instead?  This bug stems from missing the change in both places for  CASSANDRA-16064, and I can't remember why we though it was good. Pre-16064 this would have made sure the value was >=1 for a sane local clock.

Probably more for @dcapwell to remember than @yifan-c

Reply:
Any value >= 0 for the first byte in vUnsignedInt can be correctly recognized that the vUnsignedInt contains only 1 byte. Not sure how the `+1` helps. And it does not help to prevent negative values too. (except -1)
I would guess `+1` was written before 
```
// in org.apache.cassandra.utils.vint.VIntCoding#computeUnsignedVIntSize(long)
Long.numberOfLeadingZeros(value | 1); // | with 1 to ensure magntiude <= 63, so (63 - 1) / 7 <= 8
```
The latter one (`value | 1`) is better and voids the need for `+1`.

Reply:
I removed the +1 as it was missing on deserialize so caused `msg != deserialize(serialize(msg))`; it looks like I missed the removal of `+1` on size of.

I talked to Aleksey and Benedict about why the `+1` was there and neither had an answer and said to remove it.